contracts/PrepaidCardMarket_v2.sol

```diff
@@ -0,0 +1,304 @@
+pragma solidity ^0.8.9;
+pragma abicoder v1;
+
+import "./core/Ownable.sol";
+
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "./core/Versionable.sol";
+import "./IPrepaidCardMarket.sol";
+import "./PrepaidCardManager.sol";
+import "./ActionDispatcher.sol";
+import "./VersionManager.sol";
+import "./TokenManager.sol";
+import "./core/Safe.sol";
+import "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxyFactory.sol";
+import "./libraries/SafeERC677.sol";
+
+import "hardhat/console.sol";
+
+contract PrepaidCardMarketV2 is Ownable, Versionable, Safe {
+  using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+  using SafeERC20Upgradeable for IERC677;
+  using SafeERC677 for IERC677;
+
+  struct SKU {
+    address issuerSafe;
```

Comment:
`issuerSafe` was added so that we can get the safe address in the `getQuantity(bytes32 sku)` method.

I suppose an alternative would be to remove `issuer` and keep `issuerSafe` in the SKU struct, and get the issuer from the issuer mapping in other places in code where we need the issuer. What do you think?

Reply:
i think an issuer can have muliple safes that they use to fund inventory but still retain the same SKU--meaning that the issuerSafe should not be part of the SKU. If it's just a lookup that we need, then let's just create a mapping like `issuers`, only it will be in the opposite direction. Since the issuer's safe is not part of the SKU then i would prefer not to store it in this struct.

Reply:
as I look deeper into the logic i see a conundrum in that if an issuer has multiple safes, you won't know which one to pull from (ideally any might be valid--but figuring out which and being able to roll over into different safe balances could be tricky). Given that, I'm ok with having the issuer safe as part of the SKU struct, but probably we should not make it part of the SKU hash itself, so that eventually we could support the ability to have multiple safes fund all the prepaid cards for an issuer.